src/datasources/db/v2/database-migrator.service.spec.ts

```diff
@@ -0,0 +1,140 @@
+  let databaseMigratorService: DatabaseMigrator;
+  let loggingService: ILoggingService;
+  let connection: DataSource;
+  const NUMBER_OF_RETRIES = 2;
+  const truncateLockQuery = 'TRUNCATE TABLE "_lock";';
+  const insertLockQuery = 'INSERT INTO "_lock" (status) VALUES ($1);';
+
+  beforeAll(async () => {
+    // We should not require an SSL connection if using the database provided
+    // by GitHub actions
+    const isCIContext = process.env.CI?.toLowerCase() === 'true';
+    const baseConfiguration = configuration();
+    const testConfiguration: typeof configuration = () => ({
+      ...baseConfiguration,
+      db: {
+        ...baseConfiguration.db,
+        migrator: {
+          numberOfRetries: NUMBER_OF_RETRIES,
+          migrationsExecute: false,
+          retryAfter: 10,
+        },
+        connection: {
+          postgres: {
+            ...baseConfiguration.db.connection.postgres,
+            ssl: {
+              ...baseConfiguration.db.connection.postgres.ssl,
+              enabled: !isCIContext,
+              caPath: join(__dirname, '../../../../db_config/test/server.crt'),
+            },
+          },
+        },
+      },
+    });
+
+    moduleRef = await Test.createTestingModule({
+      imports: [
+        TestLoggingModule,
+        ConfigurationModule.register(testConfiguration),
+        ConfigModule,
+        TestPostgresDatabaseModuleV2,
+      ],
+      providers: [DatabaseMigrator],
+    }).compile();
+
+    databaseMigratorService = moduleRef.get<DatabaseMigrator>(DatabaseMigrator);
+    postgresDatabaseService = moduleRef.get<PostgresDatabaseService>(
+      PostgresDatabaseService,
+    );
+    loggingService = moduleRef.get<ILoggingService>(LoggingService);
+
+    connection = postgresDatabaseService.getDataSource();
+  });
+
+  afterAll(async () => {
+    await moduleRef.close();
+  });
+
+  beforeEach(() => {});
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('migrate()', () => {
+    it('Should log the start and end of the migration process', async () => {
+      jest.spyOn(loggingService, 'info');
+
+      await databaseMigratorService.migrate();
+
+      expect(loggingService.info).toHaveBeenCalledWith(
+        'Migrations: Running...',
+      );
+      expect(loggingService.info).toHaveBeenCalledWith('Migrations: Finished.');
+      expect(loggingService.info).toHaveBeenCalledTimes(3);
+    });
+
+    it('Should run migration if no lock exists', async () => {
+      jest.spyOn(connection, 'query').mockImplementation(jest.fn());
+
+      await databaseMigratorService.migrate();
+
+      expect(connection.runMigrations).toHaveBeenCalled();
+      expect(connection.query).toHaveBeenCalledWith(insertLockQuery, [1]);
+
+      expect(connection.query).toHaveBeenCalledWith(truncateLockQuery);
+    });
+
+    it('Should throw an error if retries are exhausted', async () => {
+      connection.query = jest.fn().mockResolvedValue([{ id: 1, status: 1 }]);
```

Comment:
Similar to the above, the methods are mocked already. This means we can use `mockResolvedValue` directly on the property. You might need to adjust the type as above.

Reply:
Done https://github.com/safe-global/safe-client-gateway/commit/2ea0e5d8a752fd31cff745ff3490d8fe8134ddc7